### PR TITLE
[CICD] Add Enflame unit test CI support

### PR DESCRIPTION
**Pull request by fanmeng-123**
# Description

Add CI/CD workflow support for Enflame L600 unit tests in Megatron-LM-FL.
The implementation reuses the existing configuration-driven common workflow
and adds Enflame-specific runtime setup, dependency installation, coverage
compatibility handling, and an eight-device unit test matrix.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Infra/Build change (changes to CI/CD workflows or build scripts)
- [ ] Code refactoring
- [ ] Documentation change
- [x] Bug fix
- [ ] Breaking change

## Changes

- **Enflame platform configuration**: Added the Enflame hardware configuration
  with the Enflame CI image, `sy-8g-cicd-megatron` runner label, S60 device
  type, container options, local asset mounts, and eight-process execution.

- **Enflame unit workflow**: Added the Enflame workflow based on the shared
  `all_tests_common.yml` workflow. Enflame unit test execution is enabled
  through the platform configuration, while functional test execution remains
  disabled until the functional test environment is ready.

- **Unit test matrix**: Added Enflame unit test groups covering core, models,
  distributed, distributed checkpointing, tensor parallel, pipeline parallel,
  data, fusions, export, post-training, tokenizer, and utility tests.

- **Enflame runtime setup**: Added validation for the installed PyTorch and
  `torch_gcu` versions, GCU availability, and the expected device capacity.
  The setup script also installs the unit-test dependencies and the editable
  Megatron package.

- **Coverage compatibility**: Added Enflame-specific handling for the
  `torch_gcu` coverage startup issue and cleanup of stale coverage hooks from
  previous image versions.

- **Unit runner compatibility**: Prevented helper-process output used to build
  pytest arguments from being mixed into the main command stream.

- **Platform detection coverage**: Added regression tests for Enflame runtime
  detection and normal CUDA detection behavior.

- **Workflow trigger cleanup**: Limited existing platform workflows to pull
  requests and manual execution to avoid redundant push-triggered test runs.

## Validation

- Enflame setup validates the GCU runtime and device capacity before tests run.
- The Enflame workflow uses the shared unit-test workflow and coverage reporting
  path.
- Python syntax validation passed.
- `git diff --check` passed.
- Full unit-test validation should be performed on the Enflame CI runner with
  the configured image and hardware.

## Checklist

- [x] I have read and followed the contributing guidelines
- [x] The functionality is complete
- [x] I have commented my code, particularly in CI workflow setup steps
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added/updated tests that prove my changes work
- [ ] New and existing unit tests pass locally on Enflame hardware

### CODE REVIEW

File: .github/workflows/all_tests_enflame.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: enflame_tests
+
+on:
+  push:
+    branches: ["main","dev-Enflame"]
+  pull_request:
+    branches: ["main","dev-Enflame"]
```
Comment (Darryl233) [inline .github/workflows/all_tests_enflame.yml]: Remove dev-Enflame?

File: .github/scripts/set_env_enflame.sh [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    "import torch, torch_gcu; print(torch.gcu.device_count())" |
+    awk '/^[0-9]+$/ { count = $0 } END { print count }')
+  ci_validate_device_capacity "$device_count"
+
+  python3 -c \
+    "import torch, torch_gcu; assert torch.gcu.is_available(); print(f'GCU devices: {torch.gcu.device_count()}')"
+}
+
+clean_stale_coverage_patch() {
+  # Clean up stale .pth file from earlier image builds (commit 539097af3 → 271405d79).
```
Comment (Darryl233) [inline .github/scripts/set_env_enflame.sh]: Seems this is a issue which should be solved during image building

File: .github/workflows/all_tests_enflame.yml (hunk 1 above)
Comment (fanmeng-123) [reply .github/workflows/all_tests_enflame.yml]: updated

File: .github/scripts/set_env_enflame.sh (hunk 2 above)
Comment (fanmeng-123) [reply .github/scripts/set_env_enflame.sh]: updated

File: megatron/plugin/platform/platform_cuda.py [hunk 3]
```diff
@@ -26,7 +26,7 @@ def __init__(self):
     def is_available(self):
         try:
             import torch
-            # Determine if we are on a GPU or x86 CPU with torch.
```
Comment (Darryl233) [inline megatron/plugin/platform/platform_cuda.py:29]: unnecessary change

File: megatron/plugin/tests/test_platform.py [hunk 4]
```diff
@@ -346,6 +346,7 @@ def test_get_device_capability_raises(self):
             self.cpu.get_device_capability()
 
 
+
```
Comment (Darryl233) [inline megatron/plugin/tests/test_platform.py]: unnecessary change

**Review by Darryl233 — APPROVED** [review]
LGTM

**Review by xmhubj — APPROVED** [review]
